crates/sev-veracruz-server/src/server.rs

```diff
@@ -0,0 +1,182 @@
+//! Implementation of VeracruzServer for AMD SEV-SNP
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE_MIT.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+use anyhow::{anyhow, Result};
+use policy_utils::policy::Policy;
+use proxy_attestation_client;
+use raw_fd;
+use std::{
+    error::Error,
+    os::unix::io::AsRawFd,
+    process::{
+        Child,
+        Command,
+        Stdio,
+    },
+    thread::sleep,
+    time::{
+        Duration,
+        Instant,
+    },
+};
+#[cfg(feature = "debug")]
+use std::fs::OpenOptions;
+use veracruz_server::{VeracruzServer, VeracruzServerError};
+use veracruz_utils::runtime_manager_message::{
+    RuntimeManagerRequest, RuntimeManagerResponse, Status,
+};
+use vsocket::VsockSocket;
+
+pub struct VeracruzServerSev {
+    /// A convenience struct for handling VSOCK connections to the enclave
+    vsocksocket: VsockSocket,
+    vm_process: Child,
+}
+
+impl VeracruzServer for VeracruzServerSev {
+    fn new(policy_json: &str) -> Result<Self, VeracruzServerError> {
+        println!("VeracruzServerSev::new started");
+        let policy: Policy = Policy::from_json(policy_json)?;
+        let (challenge_id, challenge) = proxy_attestation_client::start_proxy_attestation(
+            policy.proxy_attestation_server_url(),
+        )
+        .map_err(|err| {
+            eprintln!("Failed to start proxy attestation process. Error produced: {}.",
+                err,
+            );
+            err
+        })?;
+
+        let cid: u32 = 3; // TODO: Don't hard-code this
```

Comment:
TODO here? I do not see problem to hardcode, but maybe use all capital letters.

Reply:
resolved